src/components/markdown/markdown.tsx

```diff
@@ -119,9 +119,29 @@ export const Markdown = forwardRef<HTMLDivElement, MarkdownProps>(
       const componentMapWithPassedThroughProps = Object.entries(
         componentMap
       ).reduce((newMap, [key, Component]) => {
-        newMap[key] = (props: any) => (
-          <Component {...props} {...(componentProps || {})} />
-        );
+        newMap[key] = (props: any) => {
+          // Replace \r\n and \n with <br /> for paragraphs
+          const children =
+            key === 'p'
+              ? props?.children?.map((child: any, i: number) => {
+                  if (typeof child == 'string') {
+                    const lines = child.split(/\r\n|\n/g);
+                    return lines.map((str, i) => (
+                      <React.Fragment key={i}>
+                        {str}
+                        {i !== lines.length - 1 && <br />}
+                      </React.Fragment>
+                    ));
+                  }
+                  return child;
+                })
+              : props.children;
```

Comment:
Minor: We've got `props?.children` on L126 -- do we need the same safety guard here?

Reply:
If it's falsey, typically null or undef, nothing will output unless it can be coerced like string "0" or "false".

Reply:
I guess another way is asking -- can `props` ever be falsey, such that `.children` access will throw an error? If not, then we can remove the `?` from `props?.children` on L126

Reply:
Totally, I removed the safety guard from L126

Reply:
It cannot when using JSX. So if you do `<Foo />` props will be `{}`. I'm sure there is a way to construct a component using a util function to make the props undefined, but that's not typical.